quodlibet/quodlibet/formats/vgm.py

```diff
@@ -38,24 +56,84 @@ def __init__(self, filename):
 
                 self["~#length"] = length
 
+                gd3_position = struct.unpack('<i',
+                                             header[GD3_TAG_PTR_POS:
+                                                    GD3_TAG_PTR_POS
+                                                    + GD3_TAG_PTR_SIZE])[0]
+                h.seek(GD3_TAG_PTR_POS + gd3_position)
+                self.update(parse_gd3(h.read()))
+
+        self.setdefault(
+            "title", fsn2text(path2fsn(os.path.basename(filename)[:-4])))
         self.sanitize(filename)
 
     def write(self):
         pass
 
-    def reload(self, *args):
-        title = self.get("title")
-        super(VgmFile, self).reload(*args)
-        if title is not None:
-            self.setdefault("title", title)
-
     def can_change(self, k=None):
         if k is None:
             return ["title"]
         else:
             return k == "title"
 
 
+def parse_gd3(data):
+    tags = {}
+    if data[0:4] != b'Gd3 ':
+        print('Invalid Gd3, Missing Header...')
+        return tags
+
+    # version = data[4:8]
+    # Should be [0x00, 0x10, 0x00, 0x00] currently.
+    # We should hold onto it for possible branching if standards change.
+
+    gd3_length = struct.unpack('<i', data[8:12])[0]
+    # Length of gd3 footer. This means we can actually add more tags to the end
+    # with APETAG at some point. (Or at least consider...)
+
+    entries = data[12:12 + gd3_length].decode('utf-16-le').split('\0')
+
+    titles = gd3_filter_entries([entries[GD3_ENGLISH_TITLE],
```

Comment:
`entries[GD3_ENGLISH_TITLE]` can still raise IndexError

Reply:
Good point. New patch now make sure that the entries are within the bounds of the array for each element along the way.